hack/windows/Install-OVS.ps1

```diff
@@ -244,49 +421,199 @@ function InstallDependency() {
+        if (-not (Test-Path $LocalFile)){
+            Log "Path $LocalFile doesn't exist, exit"
+            exit 1
+        }
 
-if ($InstallUserspace -eq $true) {
-    InstallDependency
+        $ovsFile = Get-Item $LocalFile
+        if ($ovsFile -is [System.IO.DirectoryInfo])  {
+            return $ovsFile.FullName
+        }
 
-    ConfigOVS
+        # $ovsFile as a zip file is supported
+        $attributes = $ovsFile.Attributes
+        if (("$attributes" -eq "Archive") -and ($ovsFile.Extension -eq ".zip" ) ) {
+            $OVSZip = $LocalFile
+            $OVSDownloadURL = ""
+            $OVSPublishedHash = ""
+        } else {
+            Log "Unsupported local file $LocalFile with attributes '$attributes'"
+            exit 1
+        }
+    } else {
+        $OVSZip = "$WorkDir\ovs-win64.zip"
+        if ($DownloadURL -ne "" -and $DownloadURL -ne "$OVSDownloadURL") {
+            $OVSDownloadURL = $DownloadURL
+            $desiredOVSPublishedHash = "invalid"
+        }
+    }
+
+    # Extract zip file to $OVSInstallDir
+    if (Test-Path -Path $OVSInstallDir) {
+        Log "$OVSInstallDir already exists, exit OVS installation."
+        exit 1
+    }
+    $removeZipFile = $false
+    if ($OVSDownloadURL -ne "") {
+        DownloadOVS -localZipFile $OVSZip -downloadURL $OVSDownloadURL -desiredHash $desiredOVSPublishedHash
+        $removeZipFile = $true
+    }
+    $ovsInstallParentPath = Split-Path -Path $OVSInstallDir -Parent
+    Expand-Archive -Path $OVSZip -DestinationPath $ovsInstallParentPath | Out-Null
+    if ($removeZipFile) {
+        rm $OVSZip
+    }
+    return $OVSInstallDir
+}
+
+function CopyOVSUtilities() {
+    param (
+        [parameter(Mandatory = $true)] [string] $OVSInstallPath
+    )
+    $hostUsrBinPath = "C:\openvswitch\usr\bin"
+    $usrBinPath="${OVSInstallPath}\usr\bin"
+    if ("$usrBinPath" -ne "$hostUsrBinPath") {
+        if ((-not $global:updateDriver) -and (Test-Path $hostUsrBinPath)) {
+            return
+        }
+        if (Test-Path $hostUsrBinPath) {
+            rm -r $hostUsrBinPath
+        }
+        mkdir -p $hostUsrBinPath
+        cp -r $usrBinPath\* $hostUsrBinPath
+    }
+    AddToEnvPath($hostUsrBinPath)
 }
 
-# Antrea Pod runs as NT AUTHORITY\SYSTEM user on Windows, antrea-ovs container writes
-# pid and conf.db files to $OVSInstallDir on Windows host Node during runtime.
-icacls $OVSInstallDir /grant "NT AUTHORITY\SYSTEM:(OI)(CI)F" /T
+function InstallOVS() {
+    param (
+        [parameter(Mandatory = $true)] [string] $OVSInstallPath
+    )
+    # Install powershell modules
+    $OVSScriptsPath = "${OVSInstallPath}\scripts"
+    CheckAndInstallScripts($OVSScriptsPath)
+
+    # Install VC redistributables.
+    $OVSRedistDir="${OVSInstallPath}\redist"
+    # Check if the VC redistributable is already installed. If not installed, or the installed version
+    # is lower than $MininalVCRedistVersion, install VC redistributable files provided in the container.
```

Comment:
I don't think we have to mention the "container" here and in the comment below. We still say that this script can be run from the host.

Reply:
updated as `install the local provided VC redistributable files or download the file and install it.`